### PR TITLE
Extract common buildpack code

### PR DESCRIPTION
**Pull request by menehune23**
Introduces a common go module from which to share code that is common among buildpacks

- Moves `python/java.FuncYaml` (and tests) to common `config.FuncYaml`
- Moves `python.CommandRunner` to common `command.Runner`

TODO:
- [x] Change target branch back to `main` after #137 is merged

### CODE REVIEW

**Comment by menehune23** [comment]
@andrew-su Is there something I'm missing with the dependency review? I see similar `go.sum` entries for other modules on the `main` branch (and in other PRs, like #137) which don't fail.

**Comment by andrew-su** [comment]
I wonder if this comes from it being a new `go.mod` and `go.sum`.

**Comment by andrew-su** [comment]
From https://github.com/vmware-tanzu/function-buildpacks-for-knative/actions/runs/3184247436 (Failed dependency review run for #137) it looks like it did complain about it in the beginning, but it doesn't care about old existing problem after you pushed some more changes.

**Comment by andrew-su** [comment]
There's an issue: https://github.com/vmware-tanzu/function-buildpacks-for-knative/issues/55 for updating `kn-plugin-func` to the latest version. IIRC, they updated their code for the plugin and made it unusable as a library. Maybe things have changed and we should probably revisit that. I'd very much like to not be on such an old version.

File: buildpacks/java/tests/setup.go [hunk 1]
```diff
@@ -4,58 +4,13 @@
...
-	"log"
 	"os"
 	"path/filepath"
 
-	"github.com/paketo-buildpacks/libpak/bard"
-	knfn "knative.dev/kn-plugin-func"
+	"kn-fn/buildpacks/tests"
 )
 
-func NewLogger() bard.Logger {
```
Comment (menehune23) [inline buildpacks/java/tests/setup.go:18]: Most of this file moved to the common dir

Comment (menehune23) [reply buildpacks/java/tests/setup.go:18]: What's left is java-specific

File: buildpacks/python/tests/setup.go [hunk 2]
```diff
@@ -50,126 +46,7 @@ func (f HTTPFunction) Generate(path, funcTemplate string) error {
 	return templ.Execute(file, f)
 }
 
-func NewLogger() bard.Logger {
```
Comment (menehune23) [inline buildpacks/python/tests/setup.go:53]: Most of this file moved to the common dir. What's left is python-specific

File: buildpacks/Makefile [hunk 3]
```diff
@@ -9,11 +9,12 @@ buildpack.path := $(abspath $(path))
 include $(buildpack.path)/rules.mk
 
 # Include test case images
-buildpack.languages = \
+buildpack.subdirs = \
+    common \
```
Comment (andrew-su) [inline buildpacks/Makefile:13]: Indentation

**Review by andrew-su — APPROVED** [review]
Everything else looks 👍

File: buildpacks/Makefile (hunk 3 above)
Comment (menehune23) [reply buildpacks/Makefile:13]: Ah good catch

**Pull request merged**